module/behaviour/strategy/SoccerStrategy/src/SoccerStrategy.cpp

```diff
@@ -222,52 +222,15 @@ namespace module::behaviour::strategy {
-                                    initialLocalisationReset();
-                                }
-                                currentState = Behaviour::State::INITIAL;
-                            }
-                            else if (phase == Phase::READY) {
-                                if (gameState.data.our_kick_off) {
-                                    walkTo(fieldDescription, cfg_.start_position_offensive);
-                                }
-                                else {
-                                    walkTo(fieldDescription, cfg_.start_position_defensive);
-                                }
-                                find({FieldTarget(FieldTarget::Target::SELF)});
-                                currentState = Behaviour::State::READY;
-                            }
-                            else if (phase == Phase::SET) {
-                                standStill();
-                                find({FieldTarget(FieldTarget::Target::BALL)});
-                                if (mode == GameMode::PENALTY_SHOOTOUT) {
-                                    if (currentState != previousState) {
-                                        penaltyShootoutLocalisationReset(fieldDescription);
-                                    }
-                                    emit(std::make_unique<ResetRawSensors>());
-                                }
-                                currentState = Behaviour::State::SET;
-                            }
-                            else if (phase == Phase::TIMEOUT) {
-                                standStill();
-                                find({FieldTarget(FieldTarget::Target::SELF)});
-                                currentState = Behaviour::State::TIMEOUT;
-                            }
-                            else if (phase == Phase::FINISHED) {
-                                standStill();
-                                find({FieldTarget(FieldTarget::Target::SELF)});
-                                currentState = Behaviour::State::FINISHED;
-                            }
-                            else if (phase == Phase::PLAYING) {
-                                play(field, ball, fieldDescription, mode);
-                            }
-                        }
+
+                    // Switch gamemode statemachine based on GameController state
+                    switch (mode) {
+                        case GameMode::PENALTY_SHOOTOUT: penaltyShootout(phase, fieldDescription, field, ball); break;
+                        // We handle NORMAL and OVERTIME the same at the moment because we don't have any special
+                        // behaviour for overtime.
+                        case GameMode::NORMAL: normal(gameState, phase, fieldDescription, field, ball); break;
+                        case GameMode::OVERTIME: normal(gameState, phase, fieldDescription, field, ball); break;
```

Comment:
If we handle both cases the same then handle both cases the same
```suggestion
                        case GameMode::NORMAL: 
                        case GameMode::OVERTIME: normal(gameState, phase, fieldDescription, field, ball); break;
```